pkg/rtc/transport.go

```diff
@@ -706,6 +704,11 @@ func (t *PCTransport) onPeerConnectionStateChange(state webrtc.PeerConnectionSta
 	case webrtc.PeerConnectionStateFailed:
 		t.clearConnTimer()
 		t.handleConnectionFailed(false)
+	case webrtc.PeerConnectionStateClosed:
+		// peerconnection closed by client
+		if !t.isClosed.Load() {
+			t.params.Handler.OnClosed()
```

Comment:
pion v4 will fire `Closed` state for client close peerconnection actively and no `failed` state in this case, handle it to close the participant.